dbms/src/Functions/FunctionsRegexp.h

```diff
@@ -1512,6 +1529,349 @@ class FunctionStringRegexpInstr : public FunctionStringRegexpBase
+        {
+            if (expr_param.isNullAt(0) || pat_param.isNullAt(0) || pos_param.isNullAt(0) || occur_param.isNullAt(0) || match_type_param.isNullAt(0))
+            {
+                res_arg.column = res_arg.type->createColumnConst(col_size, Null());
+                return;
+            }
+
+            String pat = pat_param.getString(0);
+            if (unlikely(pat.empty()))
+                throw Exception(EMPTY_PAT_ERR_MSG);
+
+            int flags = getDefaultFlags();
+            String expr = expr_param.getString(0);
+            String match_type = match_type_param.getString(0);
+            pat = fmt::format("({})", pat);
+
+            Regexps::Regexp regexp(addMatchTypeForPattern(pat, match_type, collator), flags);
+            auto res = regexp.substr(expr.c_str(), expr.size(), pos_const_val, occur_const_val);
+            if (res)
+                res_arg.column = res_arg.type->createColumnConst(col_size, toField(res.value().toString()));
+            else
+                res_arg.column = res_arg.type->createColumnConst(col_size, Null());
+            return;
+        }
+
+        // Initialize result column
+        auto col_res = ColumnString::create();
+        col_res->reserve(col_size * 15);
+
+        constexpr bool has_nullable_col = ExprT::isNullableCol() || PatT::isNullableCol() || PosT::isNullableCol() || OccurT::isNullableCol() || MatchTypeT::isNullableCol();
+
+#define GET_POS_VALUE(idx)                          \
+    do                                              \
+    {                                               \
+        if constexpr (PosT::isConst())              \
+            pos = pos_const_val;                    \
+        else                                        \
+            pos = get_pos_func(pos_container, idx); \
+    } while (0);
+
+#define GET_OCCUR_VALUE(idx)                              \
+    do                                                    \
+    {                                                     \
+        if constexpr (OccurT::isConst())                  \
+            occur = occur_const_val;                      \
+        else                                              \
+            occur = get_occur_func(occur_container, idx); \
+    } while (0);
+
+        StringRef expr_ref;
+        String pat;
+        Int64 pos;
+        Int64 occur;
+        String match_type;
+
+        auto null_map_col = ColumnUInt8::create();
+        typename ColumnUInt8::Container & null_map = null_map_col->getData();
+        null_map.resize(col_size);
+
+        // Start to execute instr
+        if (canMemorize<PatT, MatchTypeT>())
+        {
+            std::unique_ptr<Regexps::Regexp> regexp;
+            if (col_size > 0)
+            {
+                regexp = memorize<true>(pat_param, match_type_param, collator);
+                if (regexp == nullptr)
+                {
+                    null_map.resize(col_size, 1);
```

Comment:
Are you sure this will work? Because in L1698, the `null_map` is already resized to `col_size`